api/src/services/graphql/subscription.ts

```diff
@@ -38,26 +43,15 @@ export function createSubscriptionGenerator(self: GraphQLService, event: string)
 
 				for (const key of keys) {
 					const data = await service.readOne(key, { fields } as Query);
-					yield { [event]: { ...data, _event: 'update' } };
+					yield { [event]: { id: key, data, event: 'update' } };
 				}
 			}
 
 			if (eventData['action'] === 'delete') {
 				const { keys } = eventData;
 
 				for (const key of keys) {
-					const result: Record<string, any> = {};
-
-					if (fields) {
-						for (const field of fields) {
-							result[field] = null;
-						}
-					}
-
-					const pk = schema.collections[eventData['collection']]?.primary;
-					if (pk) result[pk] = key;
-					result['_event'] = 'delete';
-					yield { [event]: result };
+					yield { [event]: { id: key, data: {}, event: 'delete' } };
```

Comment:
```suggestion
					yield { [event]: { id: key, event: 'delete' } };
```
If a non-null field is in the selection set, an error is thrown.
```
Cannot return null for non-nullable field a.id.
```
By removing the `data` here, the data will simply be `null`.
Currently when nullable fields are selected, the object is constructed but values are all set to `null`.

```
subscription{
  a_mutated(event:delete){
    event
    id
    data {
      name
      description
    }
  }
}

Result {
  "data": {
    "a_mutated": {
      "event": "delete",
      "id": "23",
      "data": {
        "name": null,
        "description": null
      }
    }
  }
} 
```